src/lib/index/index_service.cpp

```diff
@@ -35,22 +36,23 @@ IndexService::IndexService(std::unique_ptr< IndexServiceCallbacks > cbs) : m_svc
 }
 
 void IndexService::create_vdev(uint64_t size) {
-    auto const atomic_page_size = hs()->device_mgr()->atomic_page_size({PhysicalDevGroup::FAST});
+    auto const atomic_page_size = hs()->device_mgr()->atomic_page_size(HSDevType::Fast);
+    hs_vdev_context vdev_ctx;
+    vdev_ctx.type = hs_vdev_type_t::INDEX_VDEV;
 
-    struct blkstore_blob blob;
-    blob.type = blkstore_type::INDEX_STORE;
-    m_vdev =
-        std::make_shared< VirtualDev >(hs()->device_mgr(), "index", PhysicalDevGroup::FAST, blk_allocator_type_t::fixed,
-                                       size, 0, true, atomic_page_size, (char*)&blob, sizeof(blkstore_blob), true);
+    hs()->device_mgr()->create_vdev(vdev_parameters{.vdev_name = "index",
+                                                    .vdev_size = size,
+                                                    .num_chunks = 1,
+                                                    .blk_size = atomic_page_size,
+                                                    .dev_type = HSDevType::Fast,
+                                                    .multi_pdev_opts = vdev_multi_pdev_opts_t::ALL_PDEV_STRIPED,
+                                                    .context_data = vdev_ctx.to_blob()});
 }
 
-void IndexService::open_vdev(vdev_info_block* vb) {
-    m_vdev = std::make_shared< VirtualDev >(hs()->device_mgr(), "index", vb, PhysicalDevGroup::FAST,
-                                            blk_allocator_type_t::fixed, vb->is_failed(), true);
-    if (vb->is_failed()) {
-        LOGINFO("index vdev is in failed state");
-        throw std::runtime_error("vdev in failed state");
-    }
+shared< VirtualDev > IndexService::open_vdev(const vdev_info& vinfo, bool load_existing) {
+    m_vdev = std::make_shared< VirtualDev >(*(hs()->device_mgr()), vinfo, blk_allocator_type_t::fixed,
```

Comment:
load_existing doesn't seem to be used anywhere, do we expect the recovery PR to fix that or you want to do it in this PR? 

Same comment for BlkDataService::open_vdev

Reply:
load_existing differentiation is needed only by MetaBlkService and possibly ReplicationService. Since it is a common callback from vdev we need to define that.